src/main/java/org/mitre/synthea/modules/EncounterModule.java

```diff
@@ -44,19 +58,54 @@ public boolean process(Person person, long time) {
       Encounter encounter = person.record.encounterStart(time, EncounterType.WELLNESS.toString());
       encounter.name = "Encounter Module Scheduled Wellness";
       encounter.codes.add(ENCOUNTER_CHECKUP);
+      encounter.provider = person.getAmbulatoryProvider(time);
       encounter.codes.add(getWellnessVisitCode(person, time));
       person.attributes.put(ACTIVE_WELLNESS_ENCOUNTER, true);
       startedEncounter = true;
-    } else if (person.symptomTotal() > SYMPTOM_THRESHOLD) {
-      // add a symptom driven encounter if symptoms are severe
-      person.resetSymptoms();
-      Encounter encounter = person.record.encounterStart(time, EncounterType.WELLNESS.toString());
-      encounter.name = "Encounter Module Symptom Driven";
-      encounter.codes.add(ENCOUNTER_CHECKUP);
-      encounter.codes.add(getWellnessVisitCode(person, time));
-      person.attributes.put(ACTIVE_WELLNESS_ENCOUNTER, true);
-      startedEncounter = true;
-    }
+    } else if (person.symptomTotal() > EMERGENCY_SYMPTOM_THRESHOLD) {
```

Comment:
Lot of duplication here and it feels like there may be opportunity to extract out common logic, but this isn't a big deal.